builder/job.go

```diff
@@ -237,3 +263,45 @@ func Commit(d *daemon.Daemon, name string, c *daemon.ContainerCommitConfig) (str
 
 	return img.ID, nil
 }
+
+// inspectResponse looks into the http response data at r to determine whether its
+// content-type is on the list of acceptable content types for remote build contexts.
+// This function returns:
+//    - a string representation of the detected content-type
+//    - an io.Reader for the response body
+//    - an error value which will be non-nil either when something goes wrong while
+//      reading bytes from r or when the detected content-type is not acceptable.
+func inspectResponse(ct string, r io.ReadCloser, clen int) (string, io.ReadCloser, error) {
+	plen := clen
+	if plen <= 0 || plen > maxPreambleLength {
+		plen = maxPreambleLength
+	}
+
+	preamble := make([]byte, plen, plen)
+	rlen, err := r.Read(preamble)
+	if rlen == 0 || (err != nil && err != io.EOF) {
+		return ct, r, err
```

Comment:
if there's no content type, and no error and the body is empty (rlen==0), we'll return here.  I assume we'll  blow up later on, probably when we try to untar an empty tarball, but it might be good to add a test for that to see what happens and that we continue to get that except behavior in the future.